File: mobile/ios/RoleplayAI/App/AI/AIEngineFactory.swift
import Foundation

enum AIEngineFactory {
    static func make(engineId: String) -> AIEngine {
        switch engineId {
        case "llamacpp":
            return LlamaCppEngine()
        case "ortgenai":
            return OrtGenAIEngine()
        case "coreml":
            return CoreMLEngine()
        default:
            return MockEngine()
        }
    }
}

File: mobile/android/RoleplayAI/app/src/main/java/com/roleplayai/app/ai/AiEngineFactory.kt
package com.roleplayai.app.ai

import android.content.Context

class AiEngineFactory(private val context: Context) {
    fun create(engineId: String): AiEngine {
        return when (engineId) {
            "llamacpp" -> LlamaCppEngine(context)
            "onnx" -> OnnxEngine(context)
            "mlkit" -> MlKitEngine(context)
            "melange" -> MelangeEngine(context)
            else -> MockEngine()
        }
    }
}
